pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go

```diff
@@ -21,17 +22,35 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv"
 	"github.com/cockroachdb/cockroach/pkg/repstream/streampb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/settings"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
 	"github.com/cockroachdb/cockroach/pkg/sql/isql"
 	"github.com/cockroachdb/cockroach/pkg/sql/physicalplan"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/errors"
 	"github.com/cockroachdb/logtags"
 )
 
+var replanThreshold = settings.RegisterFloatSetting(
+	settings.TenantWritable,
+	"stream_replication.replan_flow_threshold",
+	"fraction of nodes in the producer or consumer job that would need to change to refresh the"+
+		" physical execution plan. If set to 0, the physical plan will not automatically refresh.",
+	0.1,
+)
+
+var replanFrequency = settings.RegisterDurationSetting(
+	settings.TenantWritable,
+	"stream_replication.replan_flow_frequency",
+	"frequency at which the consumer job checks to refresh its physical execution plan",
```

Comment:
@miretskiy You wrote the following in the original PR

> not sure how I feel about fraction here. Almost think that an absolute value might make more sense?  
Like: in a 10 node cluster, you'd need 2 nodes added before replanning. Probably though, even 1 node should be enough?

> Also, consider the fact that in an 11 node cluster, after 1 node was drained (and flow replanned with 10 nodes), you then execute your function, and delta/10 would be 1/10th -- below the threshold. So, you'd never replan in that case (apologies is my math is a bit off -- but you get an idea). I want to make sure  
that drain (and rolling restarts/upgrades) in general are handled appropriately.

> You can do fancy things, and use fraction if <1 and otherwise an absolute value?  
Or you can just use an integer.

My thinking here is that 1) it's ideal to make the setting agnostic to the size of the cluster-- perhaps larger clusters cycle through nodes more frequently; 2) I was following precedent set by related cluster settings `bulkio.restore.replan_flow_threshold`, `changefeed.replan_flow_threshold`, `bulkio.import.replan_flow_threshold`.

I'm inclined to keep this as is, unless you have strong objections?

Reply:
(i implemented your other suggestions)

Reply:
I'm fine either way -- twas just a suggestion.  Take it or not. up to your or other reviewers.